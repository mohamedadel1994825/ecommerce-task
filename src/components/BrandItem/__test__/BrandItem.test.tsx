import { cleanup, fireEvent, render } from "@testing-library/react-native";
import React from "react";
import { BrandItem } from "../index";

describe("BrandItem", () => {
  afterEach(cleanup);

  const item = { name: "Brand Name" };

  it("should match snapshot when not selected", () => {
    const { toJSON } = render(
      <BrandItem item={item} isSelected={false} onPress={() => {}} />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it("should match snapshot when selected", () => {
    const { toJSON } = render(
      <BrandItem item={item} isSelected={true} onPress={() => {}} />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it("should call onPress callback when pressed", () => {
    const onPress = jest.fn();
    const { getByTestId } = render(
      <BrandItem item={item} isSelected={false} onPress={onPress} />
    );
    fireEvent.press(getByTestId("brand-item")); // Fire press event directly
    expect(onPress).toHaveBeenCalled();
  });
});
