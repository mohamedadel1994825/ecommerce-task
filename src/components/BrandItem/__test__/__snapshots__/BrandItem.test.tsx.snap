// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrandItem should match snapshot when not selected 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#efefef",
      "borderRadius": 5,
      "opacity": 1,
      "padding": 10,
    }
  }
  testID="brand-item"
>
  <Text
    style={
      {
        "color": "#000000",
      }
    }
  >
    Brand Name
  </Text>
</View>
`;

exports[`BrandItem should match snapshot when selected 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#0040ff",
      "borderRadius": 5,
      "opacity": 1,
      "padding": 10,
    }
  }
  testID="brand-item"
>
  <Text
    style={
      {
        "color": "#FFFFFF",
      }
    }
  >
    Brand Name
  </Text>
</View>
`;
